agent/proxy.py

```diff
@@ -224,19 +234,16 @@ def rename_site_on_upstream(self, upstream: str, site: str, new_name: str):
         os.rename(old_site_file, new_site_file)
 
     @job("Update Site Status")
-    def update_site_status_job(self, upstream, site, status, skip_reload=True, extra_domains=None):
+    def update_site_status_job(self, upstream, site, status, extra_domains=None):
         self.update_site_status(upstream, site, status)
         if not extra_domains:
             extra_domains = []
         for domain in extra_domains:
             self.update_site_status(upstream, domain, status)
-        if skip_reload:
-            self.reload_nginx(defer=True)
-            return
-        self.generate_proxy_config()
-        self.reload_nginx(defer=False)
+        self.reload_nginx()
 
     @step("Update Site File")
+    @with_proxy_config_lock()
```

Comment:
I haven't checked all usage, but make sure that it's only locking 3 broad things: read, modify, write the config. To avoid contention.

Reply:
Yeah it's only around accessing / modifying. Re-checked.